corehq/apps/cloudcare/views.py

```diff
@@ -216,6 +218,39 @@ def _get_case(domain, case_id):
             return render(request, "cloudcare/cloudcare_home.html", context)
 
 
+class CloudcareClearUserData(View):
+
+    urlname = 'clear_user_data'
+    http_method_names = ['post']
+
+    @method_decorator(require_cloudcare_access)
+    @method_decorator(requires_privilege_for_commcare_user(privileges.CLOUDCARE))
+    def dispatch(self, request, *args, **kwargs):
+        return super(CloudcareClearUserData, self).dispatch(request, *args, **kwargs)
+
+    def post(self, request, domain):
+        couch_user = request.couch_user
+
+        if not couch_user.is_web_user:
+            # If this is called by a mobile user, it's most likely a mistake so we should
+            # not close all their cases.
+            return json_response({'status': 'fail'}, status_code=400)
+
+        case_ids = CaseAccessors(domain).get_open_case_ids_for_owner(couch_user.user_id)
+        for case_id in case_ids:
+            update_case(
+                domain,
+                case_id,
+                close=True,
+                xmlns=CLOUDCARE_CLOSE_XMLNS,
+            )
```

Comment:
this feels like a potentially expensive thing to be doing 1) synchronously in a view and 2) one case at a time.